### PR TITLE
feat(new_check): cloudwatch_log_group_no_secrets_in_logs

### LINKED ISSUE #1936: Implement a new check for secrets in CloudWatch Logs
**Is your feature request related to a problem? Please describe.**
Suggestion to implement a new check that performs the same secrets detection done for ECS task definitions/lambda code etc, but done against logs stored in CloudWatch to detect sensitive information being stored in logs in plaintext.

**Describe the solution you'd like**
It is likely that it is not feasible to pull all the log streams for each cloudwatch log group. I suggest that ~5-10 streams are pulled, and run through secrets-detector. Something like the below could be altered and incorporated into the logs_client - with the logGroupName being a variable (for log_group in self.log_groups).

```python
     stream_response = client.describe_log_streams(
        logGroupName="/aws/lambda/lambdaFnName", # Can be dynamic
        orderBy='LastEventTime',                 # For the latest events
        limit=5
        )

    for log_stream in stream_response["logStreams"]:
        logStreamName = log_stream["logStreamName"]

        response = client.get_log_events(
             logGroupName="/aws/lambda/lambdaFnName",
             logStreamName=logStreamName
        )
```

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by Fennerr**
### Context

Added a check to look for secrets in CloudWatch logs

### Description

Altered the CloudWatch Logs class to get log events for each log group. There are 2 parameters that determine the number of events that are collected. 
* number_of_log_streams_to_retrieve determines the number of log streams per log group to get. The latest log streams are retrieved
* events_per_log_stream_threshold determines the threshold for the number of events before it stops collecting more events. It will retrieve at most events_per_log_stream_threshold + events_per_log_stream_threshold-1 events. Could alter this to retrieve exactly events_per_log_stream_threshold, but it didn't seem to make a performance difference. Can discuss this further over slack if you want.
I think that these parameters should be configured in the config file.

The check will output all the events into a temp file, and run detect-secrets on them. If a secret is detected, it will then look to see if it can give more specific information on what lines the secrets were detected on for that event. I do this by using json.loads on the event message, and then doing a json.dumps with the indent parameter set to 2. If there are multiple lines after this, then the event is run through detect-secrets again to determine which lines of the secrets are being flagged on. I found this useful for logs being generated for the aws-controltower/CloudTrailLogs log group which outputs logs as json messages. 

The convert_to_cloudwatch_timestamp_format function in the check could move to the cloudwatch_service file, and the detect_secrets_scan function and potentially the SecretsDict could move to the utils file.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Fennerr** [comment]
I am a git noob, and I see that the commits that I made to the other branch are also in this branch.. Will look into this

File: prowler/providers/aws/services/cloudwatch/cloudwatch_no_secerts_in_logs/cloudwatch_no_secerts_in_logs.metadata.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "Provider": "aws",
+  "CheckID": "cloudwatch_no_secerts_in_logs",
```
Comment (jfagoagas) [inline prowler/providers/aws/services/cloudwatch/cloudwatch_no_secerts_in_logs/cloudwatch_no_secerts_in_logs.metadata.json]:
```suggestion
  "CheckID": "cloudwatch_no_secrets_in_logs",
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]

**Comment by MrCloudSec** [comment]
> I am a git noob, and I see that the commits that I made to the other branch are also in this branch.. Will look into this

I solved it, no worries!

File: prowler/providers/aws/services/cloudwatch/cloudwatch_service.py [hunk 2]
```diff
@@ -75,8 +75,15 @@ def __init__(self, audit_info):
         self.regional_clients = generate_regional_clients(self.service, audit_info)
         self.metric_filters = []
         self.log_groups = []
-        self.__threading_call__(self.__describe_metric_filters__)
+        # self.__threading_call__(self.__describe_metric_filters__)
         self.__threading_call__(self.__describe_log_groups__)
+        if "cloudwatch_no_secerts_in_logs" in audit_info.audit_metadata.expected_checks:
```
Comment (jfagoagas) [inline prowler/providers/aws/services/cloudwatch/cloudwatch_service.py]:
```suggestion
        if "cloudwatch_no_secrets_in_logs" in audit_info.audit_metadata.expected_checks:
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]

**Comment by MrCloudSec** [comment]
@Fennerr I simplified the service calls by only using the `filter_log_events ` call and setting a limit of the amount of events per log group.

**Review by MrCloudSec — DISMISSED** [review]
Thank you for this amazing contribution @Fennerr 🚀

File: prowler/providers/aws/services/cloudwatch/cloudwatch_log_group_no_secrets_in_logs/cloudwatch_log_group_no_secrets_in_logs.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                                # Will rescan just this event to get the type of secret and the line number
+                                event_detect_secrets_output = detect_secrets_scan(
+                                    log_event_data
+                                )
+                                [
+                                    log_stream_secrets[cloudwatch_timestamp].add_secret(
+                                        secret["line_number"], secret["type"]
+                                    )
+                                    for secret in event_detect_secrets_output
+                                ]
```
Comment (JeffreySouza) [inline prowler/providers/aws/services/cloudwatch/cloudwatch_log_group_no_secrets_in_logs/cloudwatch_log_group_no_secrets_in_logs.py]: Nit:
```suggestion
                                for secret in event_detect_secrets_output:
                                    log_stream_secrets[cloudwatch_timestamp].add_secret(
                                        secret["line_number"], secret["type"]
                                    )
```

**Review by MrCloudSec — APPROVED** [review]

**Review by jfagoagas — APPROVED** [review]
Go!

**Pull request merged**